CRM/Contact/BAO/ContactType.php

```diff
@@ -871,14 +869,17 @@ protected static function getAllContactTypes() {
     $cacheKey = 'all_' . $GLOBALS['tsLocale'];
     $contactTypes = $cache->get($cacheKey);
     if ($contactTypes === NULL) {
-      $contactTypes = (array) ContactType::get(FALSE)
-        ->setSelect(['id', 'name', 'label', 'description', 'is_active', 'is_reserved', 'image_URL', 'parent_id', 'parent_id:name', 'parent_id:label'])
-        ->execute()->indexBy('name');
-
+      $query = CRM_Utils_SQL_Select::from('civicrm_contact_type')
+        ->select(['id', 'name', 'label', 'description', 'is_active', 'is_reserved', 'image_URL', 'parent_id']);
```

Comment:
This line can cause problems during upgrades by trying to select a column that hasn't been added to the schema yet. I think it's safer to remove this line (which will default to `SELECT *`).

Reply:
Removed the select in 4f4c08155595d678386a99cff3f13540861ecf4c. The tests will have to be adapted though, once there are changes to the schema.